test_workspace/Nat.bosatsu

```diff
@@ -33,6 +33,23 @@ def mult(n1: Nat, n2: Nat) -> Nat:
         Succ(n2):
           Succ(mult(n1, n2).add(add(n1, n2)))
 
+def is_even(n: Nat) -> Bool:
+    def loop(n: Nat, res: Bool) -> Bool:
+        recur n:
+            case Zero: res
+            case Succ(n): loop(n, False if res else True)
+    loop(n, True)
+
+def div2(n: Nat) -> Nat:
+    def loop(n: Nat, acc: Nat):
+        recur n:
+            case Zero: acc
+            case Succ(n):
+                # (n + 1) / 2 = n/2 if n is even, else n/2 + 1
+                if is_even(n): loop(n, acc)
```

Comment:
we know that every other item is even. This algorithm is quadratic, but we can make it O(N).